crates/astria-account-monitor/src/config.rs

```diff
@@ -70,6 +89,46 @@ impl Display for Asset {
     }
 }
 
+#[derive(Debug, Clone, Serialize)]
+pub struct SequencerAssetsToMonitor(Vec<Asset>);
+impl<'de> Deserialize<'de> for SequencerAssetsToMonitor {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let value = String::deserialize(deserializer)?;
+
+        if value.is_empty() {
+            return Err(serde::de::Error::custom("empty asset list"));
+        }
+
+        let assets: Result<Vec<Asset>, _> = value
+            .split(',')
+            .map(str::trim)
+            .map(str::parse)
+            .collect::<Result<Vec<Asset>, _>>();
+
+        let assets = assets.map_err(serde::de::Error::custom)?;
+
+        let mut items = HashSet::new();
+        if !assets.iter().all(|item| items.insert(item)) {
+            return Err(serde::de::Error::custom("duplicate assets"));
+        }
+
+        Ok(Self(assets))
+    }
+}
+
+#[expect(clippy::into_iter_without_iter, reason = "iter() is not needed")]
+impl<'a> IntoIterator for &'a SequencerAssetsToMonitor {
```

Comment:
I think it might be less work if you added an inherent method on `SequencerAssetsToMonitor` and removed this trait impl.

```rust
impl SequencerAssetsToMonitor {
    pub(crate) fn iter(&self) -> impl Iterator<Item = &Asset> {
        self.0.iter()
    }
}
```